osu.Game/Overlays/Toolbar/ToolbarUserButton.cs

```diff
@@ -33,10 +34,29 @@ public ToolbarUserButton()
             Flow.Add(avatar = new Avatar());
         }
 
+        public override bool Contains(Vector2 screenSpacePos) => base.Contains(screenSpacePos) || (loginOverlay.IsVisible && loginOverlay.Contains(screenSpacePos));
+
         [BackgroundDependencyLoader]
-        private void load(APIAccess api, OsuConfigManager config)
+        private void load(APIAccess api, OsuGameBase game)
         {
             api.Register(this);
+
+            (loginOverlay = new LoginOverlay
+            {
+                Position = new Vector2(0, 1),
+                RelativePositionAxes = Axes.Y,
+                Anchor = Anchor.TopRight,
+                Origin = Anchor.TopRight,
+            }).Preload(game, Add);
+        }
+
+        protected override bool OnClick(InputState state)
+        {
+            if (!base.Contains(state.Mouse.NativeState.Position)) return false;

```

Comment:
Not happy about this. I think it could be avoided by making the loginOverlay not a child of the button, but instead putting the button _and_ the login overlay into a container.